sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -165,6 +166,7 @@ case class FileSourceScanExec(
     partitionFilters: Seq[Expression],
     optionalBucketSet: Option[BitSet],
     optionalNumCoalescedBuckets: Option[Int],
+    optionalDynamicDecideBucketing: Option[Boolean],
```

Comment:
nit: could you add this param in `metadata` for better explain output?

Reply:
Why does it need to be `Option[Boolean]`? I'd expect a simple `disableBucketScan: Boolean = false`